File: Program's_Contributed_By_Contributors/JavaScript_Programs/binarySearch.js
function binarySearch(list, item) {
    // Sort the list in ascending order
    list.sort((a, b) => a - b);

    let low = 0;
    let high = list.length;

    while (low <= high) {
        let mid = Math.floor((low + high) / 2);
        let guess = list[mid];
        if (guess === item) {
            return true;
        }
        if (guess < item) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    return false;
}

File: Program's_Contributed_By_Contributors/C++ Programs/Binary_Search.cpp
// This code is to implement Binary Search in C++

// Here elements[] refers to the array of elements in sorted order and searchkey refers to the key element we are searching

#include <bits/stdc++.h>
using namespace std;

int binarySearch(int elements[], int searchkey, int low, int high)
{

    while (low <= high)
    {
        int mid = low + (high - low) / 2;

        if (elements[mid] == searchkey)
            return mid;

        if (elements[mid] < searchkey)
            low = mid + 1;

        else
            high = mid - 1;
    }

    return -1;
}

int main(void)
{
    int elements[] = {-3, 5, 7, 12, 45, 72, 430};
    int searchkey = 7;
    int n = sizeof(elements) / sizeof(elements[0]);
    int result = binarySearch(elements, searchkey, 0, n - 1);
    if (result == -1)
        cout << "Element Not found";
    else
        cout << "Element is found at index " << result;
    return 0;
}